### ISSUE TITLE
Abstract the gitee backend to another repository as a plugin.

### ISSUE THREAD

**Issue by heming6666** [X-lab2017/grimoirelab-elk#3]
Maybe we can abstract the gitee backend in [raw](https://github.com/X-lab2017/grimoirelab-elk/blob/gitee/grimoire_elk/raw/gitee.py) and [enriched](https://github.com/X-lab2017/grimoirelab-elk/blob/gitee/grimoire_elk/enriched/gitee.py) to another repository as a plugin just like what [perceval backend for Gitee]( https://github.com/WillemJiang/grimoirelab-perceval-gitee) did.

**Comment by heming6666** [comment]
One problem is that `ELK` used [`get_connectors()`](https://github.com/X-lab2017/grimoirelab-elk/blob/fbce59b615ca2f37afb3babc9e3dc7d6be8e88d4/grimoire_elk/utils.py#L217) in [`util.py`](https://github.com/X-lab2017/grimoirelab-elk/blob/gitee/grimoire_elk/utils.py) to get the connectors of backend. And it was used in many places( e.g. [`grimoirelab-sirmordred/task.py`](https://github.com/chaoss/grimoirelab-sirmordred/blob/f3efb353e8f35447d4115bd6ae8c59f93855c596/sirmordred/task.py#L29) and [`grimoirelab-elk/elk.py`](https://github.com/X-lab2017/grimoirelab-elk/blob/fbce59b615ca2f37afb3babc9e3dc7d6be8e88d4/grimoire_elk/elk.py#L39) ).


So it maybe be difficult to build a seperate package(e.g. `grimoirelab-elk-gitee`) for gitee without making any modification to `grimoirelab-elk`. 

@WillemJiang is there any suggestion about this?  :)

**Comment by WillemJiang** [comment]
I just had a quick look at the util.py,  the get_connectors() just load some meta data of the connector. If util.py can load the connector meta data from the extension directory, grimoirelab-elk can support loading connector without knowing the Class in advance.

Now it's time to do some research on how to do it in python. 
https://packaging.python.org/guides/creating-and-discovering-plugins/

**Comment by heming6666** [comment]
Yes, if `util.py` can load all the connector meta data, it would be convenient to build seperate packages for any other backends.

> Now it's time to do some research on how to do it in python.
https://packaging.python.org/guides/creating-and-discovering-plugins/

Thank you @WillemJiang very much. I will look at this and update the progress here.

**Comment by WillemJiang** [comment]
I think we could use [the metadata approach](https://packaging.python.org/guides/creating-and-discovering-plugins/#using-package-metadata)  
to store the module and class information, then [import the module and assemble the classes dynamically]( https://www.geeksforgeeks.org/how-to-dynamically-load-modules-or-classes-in-python/?ref=rp).

**Comment by heming6666** [comment]
I have implemented the POC in this repository : [heming6666/python-plugins-test](https://github.com/heming6666/python-plugins-test) and write some describtion in README. Could you @WillemJiang  help to review the code and give some suggestiones whenever you have time? Then I can actually get started working on `grimoirelab-elk-gitee` .

**Comment by WillemJiang** [comment]
Yeah, it looks good. Now I'm thinking maybe we can put the gitee dashboard json part into the same repo.
As we need to modify the code grimoirelab-elk core code to enable the plugin loading mechanism, it's better we send a proposal to grimoirelab-elk first to get some feed back from the community.
